### PR TITLE
Refactor polaris-avatar component

### PR DESCRIPTION
**Pull request by vladucu**
### What's here

- refactors to ES6 class
- template-only components
- angle-bracket syntax

### CODE REVIEW

File: addon/templates/components/polaris-avatar.hbs [hunk 1]
```diff
@@ -1,29 +1,36 @@
...
-        font-size="26"
-        text-anchor="middle"
-      >
-        {{initials}}
-      </text>
-    </svg>
-  </span>
-{{/if}}
+<span
+  role={{this.role}}
```
Comment (andrewpye) [inline addon/templates/components/polaris-avatar.hbs]: Any reason to not just hard-code this now? 🤔

File: addon/components/polaris-avatar.js [hunk 2]
```diff
@@ -256,50 +221,58 @@ export default Component.extend({
...
-      'finalSource',
-      'hasError'
-    );
+  @computed('finalSource', 'hasError')
+  get shouldShowImage() {
+    let { finalSource, hasError } = this;
     return finalSource && !hasError;
-  }).readOnly(),
-
-  didReceiveAttrs() {
```
Comment (andrewpye) [inline addon/components/polaris-avatar.js]: Why did this disappear?

Comment (vladucu) [reply addon/components/polaris-avatar.js]: it's replaced by `resetImage` action below

File: addon/templates/components/polaris-avatar.hbs (hunk 1 above)
Comment (vladucu) [reply addon/templates/components/polaris-avatar.hbs]: ohh, it's private property
yeah, we can do 👍

**Pull request merged**